hls4ml/model/types.py

```diff
@@ -311,16 +314,20 @@ def __eq__(self, other):
         return eq
 
 
-class XnorPrecisionType(IntegerPrecisionType):
+class XnorPrecisionType(PrecisionType):
     """
     Convenience class to differentiate 'regular' integers from BNN Xnor ones
     """
 
     def __init__(self):
         super().__init__(width=1, signed=False)
 
+    def __str__(self):
+        typestring = 'uint<1>'
```

Comment:
I am truthfully not much of a fan of the typestring. What is it's function? There is not a 1:1 between a typestring and and the type. Part of me thinks this should be 'xnor' and not 'uint<1>', though this requires a few more changes downstream.

Reply:
Not 100% sure, but this may be the last leftover from the previous way of doing things. I think that it is only used in `WeightVariable.update_precision()` to set a new string format that is then not used anywhere (it was used only in `print_array_to_cpp` in the writer, but this was changed). If it doesn't hurt your eyes too much I would leave it as-is here, and have a follow up PR that removes it, `size_cpp()` and perhaps the whole `dim_names` concept as one of our usual "cleanup" PRs that we do prior to a release.